src/components/map/mapHelpersLatLong.js

```diff
@@ -445,6 +457,8 @@ UPDATE DEMES & TRANSMISSIONS
 
 const updateDemeDataColAndVis = (demeData, demeIndices, nodes, visibility, geoResolution, nodeColors, pieChart, colorBy, legendValues) => {
   const demeDataCopy = demeData.slice();
+  // get all possible values from tree rather than relying on legend (doesn't include 'undefined', or "", etc)
+  const allValues = Object.keys(countTraitsAcrossTree(nodes, [colorBy], false, false)[colorBy]);
```

Comment:
Haven't had a change to test this / read through everything, but I _think_ you'll have to sort this to keep the nice behavior where the arc colours match the order in the legend values.

Reply:
Oh! Hadn't even realised it was doing this. Ok, can do!

Reply:
Should be working now - thanks for pointing this out!